tools/cli/utils.go

```diff
@@ -524,10 +533,25 @@ func printError(msg string, err error) {
 	}
 }
 
-// ErrorAndPrint print easy to understand error msg first then error detail in a new line
-func ErrorAndPrint(msg string, err error) error {
-	printError(msg, err)
-	return err
+// PrintableError returns a printable error
+func PrintableError(msg string, err error) error {
+	var printable *printableError
+	if errors.As(err, &printable) { // already printable error type
+		printable.msg = msg + ": " + printable.msg
+	} else {
+		printable = &printableError{msg: msg, err: err}
+	}
+	return printable
+}
```

Comment:
gonna want to make a "fresh" wrapper every time, as otherwise currently this will take
```
err{err{printable{msg, err}}}
```
and turn it into
```
printable{"newmsg: oldmsg", err}
```
and that's probably not intended.

(also in principle mutating an error is very risky, as that could be happening concurrently.  only read, never modify)

Reply:
hm.  yea actually this is stranger the more I look at it.

for context (also in chat), generally the pattern with errors is:
- always wrap, build up ~infinite layers as necessary
- never mutate, because it could be accessed anywhere
- when "using" the error, either `As` to get the thing you care about and interrogate it, or iteratively Unwrap() and handle the whole chain somehow
  - generally in-error methods check their wrapped-err-chain for nested info, like if we had a `PrintedError{err{PrintedErr{err}}}.Print()` -> that `Print()` should probably include both printed-err messages somehow.
  - not being unwrap-able is reasonable if the outer-most layer supercedes all wrapped layers.  but you might still want to pull out wrapped "cause" errors.  or save that into each layer when wrapping like `fmt.Errorf` does, but generally this is hard to reuse.

Reply:
make sense. I'll always wrap